### DISCUSSION TITLE
How to navigate to another page that is not in the Left Navigation View

### DISCUSSION THREAD

**Discussion by affan848** [lepoco/wpfui#509] [Q&A]
Hi, 
I have made a new <ui: UiPage> OrderDetailsPage, but that page is not in the Left Navigation View, I want to load that page with click of a button, How can I do that?

I have tried
NavigationService.Navigate(typeof(Views.Pages.OrderDetailsPage));
But it does not navigate me to the page.

While using

Uri uri = new Uri(@"OrderDetailsPage.xaml", UriKind.Relative);
this.NavigationService.Navigate(uri);

this gives me an object reference null or the resource not found error.

Please help.

**Comment by affan848** [comment]
> Hi, I have made a new <ui: UiPage> OrderDetailsPage, but that page is not in the Left Navigation View, I want to load that page with click of a button, How can I do that?
> 
> I have tried NavigationService.Navigate(typeof(Views.Pages.OrderDetailsPage)); But it does not navigate me to the page.
> 
> While using
> 
> Uri uri = new Uri(@"OrderDetailsPage.xaml", UriKind.Relative); this.NavigationService.Navigate(uri);
> 
> this gives me an object reference null or the resource not found error.
> 
> Please help.

This is to update that I was able to do this using the following line of code

OrderDetailsPage orderDetailsPage = new OrderDetailsPage(new ViewModels.OrderDetailsViewModel());
            this.NavigationService.Navigate(orderDetailsPage);

Now, I want to pass the data from the source page to the destination page. Also, needs to change the title of the page to the newly navigated page.

**Reply by ultravelocity** [reply]
I'm still unable to do this. INavigationService.Navigate takes a page type or string.

In NavigationView's Navigate method, it only looks through items in the left navigation. I love the look of this UI, but I'm not sure to do a simple navigation from a list to detail page.

**Reply by ultravelocity** [reply]
I ended up just handling this myself, where the detail page is a control that gets displayed on top of my list page when an element is selected. Works well. I used XamlFlair to add a similar FadeInAndSlideUp animation.

**Comment by FloresDWMbat** [comment]
If you are using a singleton take a look on this, maybe you need to get service in right way ..

App.GetService<INavigationService>().Navigate(typeof(DashboardPage));

**Comment by lostluna1** [comment]
In latest WpfUi, you can :
```c#
App.GetRequiredService<INavigationService>().Navigate(typeof(OrderPage));
```
